### PR TITLE
scheduling/watcher, storage: store rules into the unified storage

### LINKED ISSUE #7200: mcs: support placement rules in scheduling server
## Development Task

Now, we can watch the change of placement rules with `ruleWatcher`, but we don't use it in the scheduling server with `GetRuleStorage`, which should be supported.

BTW, after the change of placement rules, we also need to call `AddSuspectKeyRange`, which is similar to `/config/rule [post]`, `/config/rule/{group}/{id} [delete]`, `/config/rule_group [post]`, `/config/rule_group/{id} [delete]`

### PR DESCRIPTION
**Pull request by JmPotato**
### What problem does this PR solve?

Issue Number: Close #7200.

### What is changed and how does it work?

```commit-message
Store rules into the unified storage rather than the isolated implementation.
```

### Check List

Tests

- Unit test
- Integration test

### Release note

```release-note
None.
```

### CODE REVIEW

**Review by lhy1024 — APPROVED** [review]

File: pkg/mcs/scheduling/server/server.go [hunk 1]
```diff
@@ -479,7 +479,7 @@ func (s *Server) startWatcher() (err error) {
 	if err != nil {
 		return err
 	}
-	s.ruleWatcher, err = rule.NewWatcher(s.Context(), s.GetClient(), s.clusterID)
+	s.ruleWatcher, err = rule.NewWatcher(s.Context(), s.GetClient(), s.clusterID, s.storage)
```
Comment (lhy1024) [inline pkg/mcs/scheduling/server/server.go:482]: BTW, after the change of placement rules, we also need to call `AddSuspectKeyRange`, which is similar to `/config/rule [post]`, `/config/rule/{group}/{id} [delete]`, `/config/rule_group [post]`, `/config/rule_group/{id} [delete]`

Comment (JmPotato) [reply pkg/mcs/scheduling/server/server.go:482]: I will implement this in a separate PR.

**Review by rleungx — APPROVED** [review]

File: pkg/mcs/scheduling/server/server.go (hunk 1 above)
Comment (JmPotato) [reply pkg/mcs/scheduling/server/server.go:486]: Implemented in https://github.com/tikv/pd/pull/7213.

**Pull request merged**